### PR TITLE
Allow duplicate dimensions and mappings

### PR DESCRIPTION
**Pull request by daniel-thom**
Completes JIRA Story DSGRID-238.

This also 
1. Cleans up some minor problems from #117 
2. Automates registration of StandardScenarios

### CODE REVIEW

File: dsgrid/config/dimension_associations.py [hunk 1]
```diff
@@ -56,7 +56,6 @@ def load(cls, path: Path, association_files):
                     .drop(column)
                     .withColumnRenamed(tmp, column)
                 )
-                table.cache()
```
Comment (elainethale) [inline dsgrid/config/dimension_associations.py:59]: Just curious--what was .cache() doing and why isn't it needed any more?

File: dsgrid/utils/spark.py [hunk 2]
```diff
@@ -118,8 +118,9 @@ def _read_natively(filename):
 
 
 def _post_process_dataframe(df, cache=False, require_unique=None):
-    if cache:
-        df = df.cache()
+    # TODO This is causing Spark warning messages. Disable until we know why.
+    # if cache:
+    #     df = df.cache()
```
Comment (elainethale) [inline dsgrid/utils/spark.py:123]: Okay--this explains why you removed .cache() above (and I googled and caught up a little on spark dataframe.cache() and .persist()).

**Review by elainethale — APPROVED** [review]
Asked and answered my own question. Looks good to me.

**Review by lixiliu — COMMENTED** [review]
I have one comment

File: dsgrid/registry/dimension_mapping_registry_manager.py [hunk 3]
```diff
@@ -79,40 +79,30 @@ def dimension_manager(self, val: DimensionRegistryManager):
...
-                raise DSGDuplicateValueRegistered(
-                    f"There are {len(duplicates)} dimension mappings with duplicate content (data files)."
+            hashes[mapping.file_hash] = mapping
+
+        existing_ids = set()
+        for i, mapping in enumerate(config.model.mappings):
+            existing = hashes.get(mapping.file_hash)
+            if (
+                existing is not None
+                and mapping.from_dimension.dimension_type == mapping.from_dimension.dimension_type
```
[the remark this answers is not in the document]
Comment (daniel-thom) [reply dsgrid/registry/dimension_mapping_registry_manager.py:93]: It's a bug...I started letting vscode make predictions and don't always pay attention. I'll fix it. thanks.

File: dsgrid/registry/dimension_registry_manager.py [hunk 4]
```diff
@@ -68,81 +68,47 @@ def name():
...
+                    replace_dim = True
+                    for field in type(dim).__fields__:
+                        exclude = set(("description", "dimension_id"))
+                        if field not in exclude and getattr(dim, field) != getattr(
+                            time_dim, field
+                        ):
+                            replace_dim = False
+            elif dim.file_hash in hashes:
+                existing = hashes[dim.file_hash]
+                if dim.dimension_type == existing.dimension_type:
```
Comment (lixiliu) [inline dsgrid/registry/dimension_registry_manager.py:97]: what if existing is None, will it throw an error here?

**Review by lixiliu — COMMENTED** [review]
one more minor comment

**Pull request merged**